File: src/main/kotlin/dev/jefy/connectpro/management/appliacaion/command/AwardCommand.java
package dev.jefy.connectpro.management.appliacaion.command;


import org.jspecify.annotations.NullMarked;

import dev.jefy.connectpro.management.appliacaion.dtos.AwardRequest;
import dev.jefy.connectpro.management.domain.vo.AwardId;

/**
 * @author Jôph Yamba
 */
@NullMarked
public interface AwardCommand {
    AwardId create(AwardRequest request);
    AwardId update(AwardId awardId, AwardRequest request);
    void delete(AwardId awardId);
}

File: src/main/kotlin/dev/jefy/connectpro/management/appliacaion/command/AwardCommand.kt
package dev.jefy.connectpro.management.appliacaion.command

import dev.jefy.connectpro.management.appliacaion.dtos.AwardRequest
import dev.jefy.connectpro.management.domain.vo.AwardId
import org.jspecify.annotations.NullMarked


/**
 * @author Jôph Yamba
 */

interface AwardCommand {
    fun create(request: AwardRequest): AwardId
    fun update(awardId: AwardId, request: AwardRequest): AwardId
    fun delete(awardId: AwardId)
}

